### DISCUSSION TITLE
Win10 Midi driver

### DISCUSSION THREAD

**Discussion by WilliamNy** [hydrogen-music/hydrogen#1495] [Hydrogen on Windows]
Hi! First, kudos to all who made this machine possible! I'm using v 1.0.1 in Win 10 which works great with the internal sequencer. I want to use my M-Audio Keystudio, which works fine with other Windows apications. The Hydrogen default midi driver is PortMidi, which I understand from the other discussions does not work, and it is the only driver supported. I've tried unsuccessfully to install 3rd party midi drivers but Windows doesn't seem to recognize them. Any suggestions? Should I try a previous release? Will it load the projects I've already saved? Thanks!!

**Comment by theGreatWhiteShark** [comment]
Could you try the latest [1.1.1](https://github.com/hydrogen-music/hydrogen/releases/tag/1.1.1) version of Hydrogen? A lot was done on the PortMIDI support and your issues might have been fixed as well.

**Reply by WilliamNy** [reply]
Ok, thanks, I will try it!

**Comment by nunosga** [comment]
WilliamNy, did you get it to work? 
I'm struggling witha similar issue. I'm trying to use a midi keyboard (Korg Microkey) to play / record patterns in Hydrogen, but i can't get Hydrogen to work with the keyboard. (the Korg works fine with the studio one DAW).

**Comment by BigD71** [comment]
Dear Sirs,
i have the same problems with Windows 10 64Bit. My MIDI-Device is a Donner DEP-20 Piano. My Piano works well on other DAW's but would not be recognized by Hydrogen. I've tried every available 64 and 32 Bit versions, but no luck at all. SEQ24 uses also PortMIDI and is working well.
Please give me an advise.

Kind regards

**Comment by theGreatWhiteShark** [comment]
Hey @nunosga @BigD71,

Is the device showing up in the [MIDI tab](http://hydrogen-music.org/documentation/manual/manual_en.html#chpt.preferences.midi_tab) of the Preferences Dialog?

Could you set the log level to Debug and post Hydrogens log in here?

Alternatively, could you try the latest development version of Hydrogen and check whether this problem still occurs? ([64 bit](https://ci.appveyor.com/api/buildjobs/o14ni3ph689uh4c2/artifacts/Hydrogen-1.1.1-win64.exe) , [32bit](https://ci.appveyor.com/api/buildjobs/x40ccciw2ltm3dd8/artifacts/Hydrogen-1.1.1-win32.exe))

**Reply by nunosga** [reply]
The keyboard shows in the midi device  list, but hydrogen does not recognise its inputs (no midi "light" in the dashboard when i press the keys)
I'll now  try the version you suggest and let you know

**Reply by theGreatWhiteShark** [reply]
> The keyboard shows in the midi device list, but hydrogen does not recognise its inputs (no midi "light" in the dashboard when i press the keys)

That's certainly strange. Would be good to see whether they appear in the log.

> I'll now try the version you suggest and let you know

Nice.

**Comment by nunosga** [comment]
now i get the midi light whenever i press the keys!!! nice!
But how do I get the keys to trigger sounds?

**Reply by theGreatWhiteShark** [reply]
> But how do I get the keys to trigger sounds?

Whenever you press a key on your MIDI keyboard a MIDI event with a specific channel (determined by some setting/preset) and specific note (determined by the key) is sent. You have to match these channels/keys with the settings of your instruments as Hydrogen does not know itself which key should trigger which instrument. In the [Instrument Editor](http://hydrogen-music.org/documentation/manual/manual_en.html#chpt.instrument_editor.general) set **Channel** to `off` and cycle through different **Note** settings while hitting a key until the instrument get's triggered. (Or have a look into Hydrogen's log to see the channel/key info of incoming MIDI events).

**Reply by nunosga** [reply]
ok!!!. Thanks for all your help!

**Comment by WilliamNy** [comment]
I downloaded the 1.1.1 version but it didn't make any difference. I gave up on that and just kept using the pattern sequencer. 

Sent from Yahoo Mail on Android 
 
  On Sun, Feb 20, 2022 at 12:10 AM, ***@***.***> wrote:   


Could you try the latest 1.1.1 version of Hydrogen? A lot was done on the PortMIDI support and your issues might have been fixed as well.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by BigD71** [comment]
Hey @theGreatWhiteShark,
i have downloaded the linked Hydrogen 64Bit Version. Here are the results:

Livingroom PC 1:
The new version worked on this PC, the connected MIDI-Devices are now listed. So everything is fine on this PC for the moment. But my Piano is connected to my Laptop.

Laptop for Music:
On my Laptop NO MIDI-Devices are listed in the linked 64Bit version.

I will attach the requested logfile and a screenshot. Thiy may help you to identify the problem. If you have any ideas, i will try them.

Kind regards


![Audio-Engine-Info](https://user-images.githubusercontent.com/115177021/194747734-00216892-c3fc-4584-8d86-f0ca27de6bd9.jpg)
[hydrogen.log](https://github.com/hydrogen-music/hydrogen/files/9741143/hydrogen.log)

**Comment by cme** [comment]
From the log it looks like Hydrogen thinks it should be using the ALSA driver, which is odd since that's not available on Windows, I wonder if default selection needs to be improved, I'll check.

But you should be able to make it work by explicitly selecting the PortMidi driver in Preferences > MIDI system > Driver > PortMidi ?

**Reply by BigD71** [reply]
Thank's for your replay. Finally I got it working. This is how:

1. Open Preferences > Audiosystem
2. Select Audio-System as Port Audio
3. Select Host-Api as MME
4. Select Audiodevice as MicrosoftSoundmapper
5. IMPORTENT: CLOSE HYDROGEN COMPLETE!
6. Start Hydrogen
7. Open Preferences > Midi System
8. Set driver to PortMIDI
9. RESTART HYDROGEN AGAIN!
10. Now my MIDI-Devices are listed and working

Kind regards

**Reply by theGreatWhiteShark** [reply]
@BigD71 thanks for the thorough analysis! These restarts are a nuisance. We will check whether they can be avoided (or at least indicate that they are required)

@cme how about we add a more general Host-API option - like `all` - which makes Hydrogen search for devices using all Host-APIs and display them as `DEVICE_NAME (HOST-API)`. Selecting a specific device will also set the Host-API under the hood.
